packages/@sfx/sayt-driver-plugin/src/sayt-driver-plugin.ts

```diff
@@ -134,14 +237,26 @@ export default class SaytDriverPlugin implements Plugin {
   }
 }
 
+export interface RequestConfig {
+  query: string;
+  searchbox?: string;
+  config?: object;
```

Comment:
Since `config` is being overridden in the subtypes and you're not expecting a `RequestConfig` anywhere else, you don't need to include it here. Had there been a need for `config` in this interface, then the better approach would be to make `RequestConfig` generic whose type parameter is the type of `config`.

Reply:
I'm interested in your reasoning as to why `config?` shouldn't be in this interface. The listed attributes are indeed what we are expecting (whether mandatory or optional) in configuration for a request. Whether or not they are extended or used generically seems somewhat irrelevant.

The other argument I can see is that this interface exists entirely to remove duplication from the other two interfaces. From that point of view, it doesn't belong here. That being said, once it's created, it need not be the 'intent' of the interface. We can take the (despicable) post-modern point of view.

Reply:
Given the context, I think having `config?` here is correct (ignore my previous judgement), but it is more correctly typed with a generic instead of `object`.

```typescript
export interface RequestConfig<T> {
  query: string;
  searchbox?: string;
  config?: T;
}
```

The other interfaces can extend it like so:

```typescript
export interafce AutocompleteRequestConfig extends RequestConfig<QueryTimeAutocompleteConfig> {}
```